great_expectations/expectations/metrics/column_map_metrics/column_values_unique.py

```diff
@@ -62,20 +62,21 @@ def _sqlalchemy_window(cls, column, _table, **kwargs):
         # the column we will be performing the expectation on, and the query is performed against it.
         dialect = kwargs.get("_dialect", None)
         sql_engine = kwargs.get("_sqlalchemy_engine", None)
-        if sql_engine and dialect and dialect.dialect.name == "mysql":
-            temp_table_name = f"ge_tmp_{str(uuid.uuid4())[:8]}"
-            temp_table_stmt = "CREATE TEMPORARY TABLE {new_temp_table} AS SELECT tmp.{column_name} FROM {source_table} tmp".format(
-                new_temp_table=temp_table_name,
-                source_table=_table,
-                column_name=column.name,
-            )
-            sql_engine.execute(temp_table_stmt)
-            dup_query = (
-                sa.select([column])
-                .select_from(sa.text(temp_table_name))
-                .group_by(column)
-                .having(sa.func.count(column) > 1)
-            )
+        if sql_engine and dialect:
+            if hasattr(dialect, "dialect") and dialect.dialect.name == "mysql":
+                temp_table_name = f"ge_tmp_{str(uuid.uuid4())[:8]}"
```

Comment:
This is the bugfix for #2978 and #2959. This is caused by the quirk in how the BigQueryDialect is imported

Reply:
nice work!